set.mm

```diff
@@ -775572,6 +775602,125 @@ additional condition (deduction form).  See ~ ralbidc for a more
+    $( Variant of ~ mof0 with stronger requirements on distinct variables.
+       (Contributed by Zhi Wang, 19-Sep-2024.) $)
+    mof0w $p |- E* f f : A --> (/) $=
+      ( vg c0 cv wf wmo wa wceq wi wal f00 simplbi eqtr3 syl2an gen2 feq1 mpbir
+      mo4 ) ADBEZFZBGUAADCEZFZHTUBIZJZCKBKUEBCUATDIZUBDIZUDUCUAUFADIZATLMUCUGUH
+      AUBLMTUBDNOPUAUCBCADTUBQSR $.
+  $}
+
+  ${
+    $d A g $.  $d f g $.  $d B f $.
+    $( There is at most one function to the empty set.  (Contributed by Zhi
+       Wang, 19-Sep-2024.) $)
+    mof0 $p |- E* f f : A --> (/) $=
+      ( vg c0 cv wf wmo wceq wi wal wex 0ex eqeq2 imbi2d albidv f00 simplbi mpg
+      spcev df-mo mpbir ) ADBEZFZBGUCUBCEZHZIZBJZCKZUCUBDHZIZUHBUGUJBJCDLUDDHZU
+      FUJBUKUEUIUCUDDUBMNOSUCUIADHAUBPQRUCBCTUA $.
+
+    $( A variant of ~ mof0 .  (Contributed by Zhi Wang, 20-Sep-2024.) $)
+    mof02 $p |- ( B = (/) -> E* f f : A --> B ) $=
+      ( c0 wceq cv wf wmo mof0 feq3 mobidv mpbiri ) BDEZABCFZGZCHADNGZCHACIMOPC
+      BDANJKL $.
+  $}
+
+  ${
+    $d A f g x $.  $d B f g x $.  $d f ph $.
+    eufsn.1 $e |- ( ph -> B e. W ) $.
+    ${
+      eufsnlem.2 $e |- ( ph -> ( A X. { B } ) e. V ) $.
+      $( There is exactly one function to a singleton.  For a simpler
+         hypothesis, see ~ eufsn assuming ~ ax-rep , or ~ eufsn2 assuming
+         ~ ax-pow and ~ ax-un .  (Contributed by Zhi Wang, 19-Sep-2024.) $)
+      eufsnlem $p |- ( ph -> E! f f : A --> { B } ) $=
+        ( vg csn cv wf wceq wb wal wex weu cxp wcel syl fconst2g alrimiv bibi2d
+        eqeq2 albidv spcedv eu6im ) ABCJZDKZLZUIIKZMZNZDOZIPUJDQAUNUJUIBUHRZMZN
+        ZDOIEUOHAUQDACFSUQGBCFUIUATUBUKUOMZUMUQDURULUPUJUKUOUIUDUCUEUFUJDIUGT
+        $.
+    $}
+
+    eufsn.2 $e |- ( ph -> A e. V ) $.
+    $( There is exactly one function to a singleton, assuming ~ ax-rep .  See
+       ~ eufsn2 for different axiom requirements.  (Contributed by Zhi Wang,
+       19-Sep-2024.) $)
+    eufsn $p |- ( ph -> E! f f : A --> { B } ) $=
+      ( vx cvv wcel csn cxp cmpt fconstmpt mptexg eqeltrid syl eufsnlem ) ABCDJ
+      FGABEKZBCLMZJKHTUAIBCNJIBCOIBCEPQRS $.
+
+    $( There is exactly one function to a singleton, assuming ~ ax-pow and
+       ~ ax-un .  Variant of ~ eufsn .  (Contributed by Zhi Wang,
+       19-Sep-2024.) $)
+    eufsn2 $p |- ( ph -> E! f f : A --> { B } ) $=
+      ( cvv wcel csn cxp snex xpexg sylancl eufsnlem ) ABCDIFGABEJCKZIJBQLIJHCM
+      BQEINOP $.
+  $}
+
+  ${
+    $d A f g y $.  $d B f g x y $.  $d V f g $.  $d Y f $.
+    $( At most one function to a singleton, with less axiom assumption than
+       ~ mofsn .  See also ~ mofsn2 .  (Contributed by Zhi Wang,
+       19-Sep-2024.) $)
+    mofsn $p |- ( B e. V -> E* f f : A --> { B } ) $=
+      ( vg wcel csn cv wf wa wceq wal wmo cxp fconst2g biimpd eqtr3 a1i syl2and
+      wi alrimivv feq1 mo4 sylibr ) BDFZABGZCHZIZAUFEHZIZJUGUIKZTZELCLUHCMUEULC
+      EUEUHUGAUFNZKZUJUIUMKZUKUEUHUNABDUGOPUEUJUOABDUIOPUNUOJUKTUEUGUIUMQRSUAUH
+      UJCEAUFUGUIUBUCUD $.
```

Comment:
```suggestion
    $( There is at most one function to a singleton, with fewer axioms than
       ~ FIX .  See also ~ mofsn2 .  (Contributed by Zhi Wang,
       19-Sep-2024.) $)
    mofsn $p |- ( B e. V -> E* f f : A --> { B } ) $=
      ( vg wcel csn cv wf wa wceq wal wmo cxp fconst2g biimpd eqtr3 a1i syl2and
      wi alrimivv feq1 mo4 sylibr ) BDFZABGZCHZIZAUFEHZIZJUGUIKZTZELCLUHCMUEULC
      EUEUHUGAUFNZKZUJUIUMKZUKUEUHUNABDUGOPUEUJUOABDUIOPUNUOJUKTUEUGUIUMQRSUAUH
      UJCEAUFUGUIUBUCUD $.
```
the theorem refers to itself so I just replaced it with FIX, though I think you meant mofsn2?

also + "there is" for mofsn2 and mofmo

Reply:
Fixed in https://github.com/metamath/set.mm/pull/4220/commits/d8e4c3466281d8160c9b92da59ee97b2bdd4cb51